src/HttpClient.ts

```diff
@@ -132,6 +132,36 @@ export class HttpClient {
             protocol: proxyConfig.protocol,
         } as AxiosProxyConfig;
     }
+    /**
+     * Performs polling on the provided URL for a given time interval.
+     * @param interval - The time interval in milliseconds at which the URL should be polled.
+     * @param url - The URL to poll.
+     * @param duration - The total duration of the polling process in milliseconds.
+     * @returns A Promise that resolves when the polling is completed.
+     */
+    public async pollURLForTime(interval: number, url: string, duration: number): Promise<IClientResponse | undefined> {
+        let counter: number = 1;
+        const endTime: number = Date.now() + duration;
+        const request: IRequestParams = {
+            url: url,
+            method: 'GET',
+        };
+        let response: IClientResponse | undefined;
+        this.logger?.debug(`Start polling...`);
+
+        while (Date.now() < endTime) {
+            try {
+                response = await this.doRequest(request);
+                this.logger?.debug(`Polling ended successfully from ${url} with status ${response.status}`);
+                return response;
+            } catch (error) {
+                this.logger?.debug(`Retry #${counter}`, JSON.stringify(error));
+            }
+            counter++;
+            await setTimeout(interval);
+        }
+        return response;
+    }
```

Comment:
I modified the default number of retry attempts from 3 to 1. I decided not to use Axios-retry for polling because I want to have control over the wait interval for each retry.